lgc/patch/PatchInOutImportExport.cpp

```diff
@@ -2832,121 +2832,65 @@ void PatchInOutImportExport::patchTcsBuiltInOutputExport(Value *output, unsigned
 
     break;
   }
-  case BuiltInTessLevelOuter: {
-    if (builtInUsage.tessLevelOuter) {
-      // Extract tessellation factors
-      std::vector<Value *> tessFactors;
-      if (!elemIdx) {
-        assert(outputTy->isArrayTy());
-
-        unsigned tessFactorCount = 0;
-
-        auto primitiveMode = m_pipelineState->getShaderModes()->getTessellationMode().primitiveMode;
-        switch (primitiveMode) {
-        case PrimitiveMode::Isolines:
-          tessFactorCount = 2;
-          break;
-        case PrimitiveMode::Triangles:
-          tessFactorCount = 3;
-          break;
-        case PrimitiveMode::Quads:
-          tessFactorCount = 4;
-          break;
-        default:
-          llvm_unreachable("Should never be called!");
-          break;
-        }
-
-        for (unsigned i = 0; i < tessFactorCount; ++i) {
-          Value *elem = ExtractValueInst::Create(output, {i}, "", insertPos);
-          tessFactors.push_back(elem);
-        }
-
-        if (primitiveMode == PrimitiveMode::Isolines) {
-          assert(tessFactorCount == 2);
-          std::swap(tessFactors[0], tessFactors[1]);
-        }
-      } else {
-        assert(outputTy->isFloatTy());
-        tessFactors.push_back(output);
+  case BuiltInTessLevelOuter:
+  case BuiltInTessLevelInner: {
+    // Extract tessellation factors
+    std::vector<Value *> tessFactors;
```

Comment:
nit: use `SmallVector<Value *>`. The llvm coding style has been revised and now suggests using `SmallVector` with the default size by default, instead of explicitly-sized `SmallVector`s or `std::vector`s: https://llvm.org/docs/ProgrammersManual.html#llvm-adt-smallvector-h

Reply:
Done.

Reply:
@xuechen417 seems like this causes function argument type mismatch. Sticking to `std::vector` is fine to keep this PR shorter, but it would be nice to make that function take something like `ArrayRef` instead of a vector reference.

Reply:
I forgot to update the function argument in last push. I used in `SmallVectorImpl` as parameter type.